src/HTTP/Exchange/Request.cs

```diff
@@ -1,3 +1,4 @@
+using System.Reflection;
```

Comment:
There's a lot of constructors in this class. The builder pattern might be useful here